src/univ/textctrl.cpp

```diff
@@ -1195,6 +1192,13 @@ void wxTextCtrl::Replace(wxTextPos from, wxTextPos to, const wxString& text)
         // update the (cached) last position first as refresh functions use it
         m_posLast += text.length() - to + from;
 
+#if defined(__WXMSW__)
+        // Take into account that every new line mark occupies
+        // two characters, not one.
```

Comment:
There really shouldn't be any reason for this to be the case with wxUniv... I.e. we should use `\n` and not `\r\n` as line delimiters.